### PR TITLE
CONSOLE-3624: Increase proxy dialer timeout to 5 minutes.

### PR DESCRIPTION
**Pull request by TheRealJon**
Increase backend proxy timeout to match browser maximum so that frontend request timeouts take precedence.

### CODE REVIEW

**Review by jhadvig — APPROVED** [review]
/lgtm

@spadgett pointing this out. If we decide to hardcode it the it looks good. Adding hold in case we want to revisit the idea of making the value configurable.
/hold

File: pkg/proxy/proxy.go
Comment (spadgett) [inline pkg/proxy/proxy.go:64]: It looks like `Dial` is now deprecated in favor of `DialContext`. Should we switch?

https://pkg.go.dev/net/http@go1.20.3#Transport

Comment (jhadvig) [reply pkg/proxy/proxy.go:64]: @spadgett yeah lets switch to the new one 👍

**Comment by TheRealJon** [comment]
@jhadvig I think we discussed hard-coding this to unblock some people, then revisiting this later as a customizable property.

**Review by jhadvig — APPROVED** [review]
/lgtm

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
QE Approver:
 /assign @yapei
Docs Approver:
 /assign @opayne1
PX Approver:
/assign @RickJWagner

**Comment by yapei** [comment]
It looks like we need some updates about increasing HTTP server response timeout, see our internal slack discussion

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

**Comment by TheRealJon** [comment]
/hold

Still figuring this one out. This might not be fully solved.

**Comment by TheRealJon** [comment]
I will be making further changes to this per [this Slack discussion](https://redhat-internal.slack.com/archives/C6A3NV5J9/p1686559711335979)

**Comment by TheRealJon** [comment]
/hold cancel

I think these are all the changes we need to make to our backend. Additional changes are required for the console operator to fully solve this.

**Comment by TheRealJon** [comment]
QE Approver
/assign @yanpzhan

File: pkg/proxy/proxy.go
Comment (jhadvig) [inline pkg/proxy/proxy.go:71]: 👍

File: pkg/proxy/proxy.go
Comment (jhadvig) [inline pkg/proxy/proxy.go:22]: nit:
```suggestion
	dialerKeepAlive       = 30 * time.Second
```

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by TheRealJon** [comment]
After further investigation, I found that these changes likely don't work as intended. The timeout defined for `Transport.DialContext` only affects connection time, not response time. Response timeout is determined by `Transport.ResponseHeaderTimeout`, which defaults to `0`. We do not explicitly set this property on our reverse proxy, so it will not timeout waiting for a response.



Docs for the packages and properties in question:
[httputil.ReverseProxy.Transport](https://pkg.go.dev/net/http/httputil#ReverseProxy:~:text=//%20The%20transport%20used%20to%20perform%20proxy%20requests.%0A%09//%20If%20nil%2C%20http.DefaultTransport%20is%20used.%0A%09Transport%20http.RoundTripper)
[http.Transport.DialContext](https://pkg.go.dev/net/http#Transport:~:text=//%20ResponseHeaderTimeout%2C%20if%20non%2Dzero%2C%20specifies%20the%20amount%20of%0A%09//%20time%20to%20wait%20for%20a%20server%27s%20response%20headers%20after%20fully%0A%09//%20writing%20the%20request%20(including%20its%20body%2C%20if%20any).%20This%0A%09//%20time%20does%20not%20include%20the%20time%20to%20read%20the%20response%20)
[net.Dialer.DialContext](https://pkg.go.dev/net#pkg-types:~:text=The%20provided%20Context%20must%20be%20non%2Dnil.%20If%20the%20context%20expires%20before%20the%20connection%20is%20complete%2C%20an%20error%20is%20returned.%20Once%20successfully%20connected%2C%20any%20expiration%20of%20the%20context%20will%20not%20affect%20the%20connection.)

**Pull request closed without merging**
